url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/TestTeaLibrary.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Scripty Programming Language
+ * Copyright (C) 2010-2012 Bruno Ranschaert, S.D.I.-Consulting BVBA
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.cmdlib;
+
+import com.sdicons.scripty.ExtensionException;
+import com.sdicons.scripty.ProcessorException;
+import com.sdicons.scripty.ScriptyStreamProcessor;
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestTeaLibrary
+{
+    private ScriptyStreamProcessor scripty;
+
+    @Before
+    public void initialize()
+    throws ExtensionException
+    {
+        scripty = new ScriptyStreamProcessor();
+        scripty.addLibraryClasses(TeaLibrary.class);
+    }
+
+    @Test
+    public void encryptStandard()
+    throws ProcessorException
+    {
+        Object lResult = scripty.process("tea-encrypt \"Bruno Ranschaert is my name\"");
+        Assert.assertEquals("58eac04cbf4629f6b576c80570e0e1898e96a6105f0d3b2780b570873f21580e601f30ab6d365853601f30ab6d365853601f30ab6d365853", lResult);
+    }
+
+    @Test
+    public void decryptStandard()
+    throws ProcessorException
+    {
+        Object lResult = scripty.process("tea-decrypt 58eac04cbf4629f6b576c80570e0e1898e96a6105f0d3b2780b570873f21580e601f30ab6d365853601f30ab6d365853601f30ab6d365853");
+        Assert.assertEquals("Bruno Ranschaert is my name", lResult);
+    }
+
+    @Test
+    public void encryptPwd()
+    throws ProcessorException
+    {
+        Object lResult = scripty.process("tea-encrypt \"Bruno Ranschaert is my name\" password=\"Lang leve de koning!\"");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)